src/main/java/com/icthh/xm/ms/entity/service/impl/XmEntityServiceImpl.java

```diff
@@ -386,10 +386,15 @@ public XmEntity findOne(IdOrKey idOrKey) {
     @Override
     @Transactional(readOnly = true)
     public XmEntity findOne(IdOrKey idOrKey, List<String> embed) {
+        final Long entityId;
         if (idOrKey.isKey()) {
-            throw new IllegalArgumentException("Key mode is not supported yet");
+            XmEntityIdKeyTypeKey projection = getXmEntityIdKeyTypeKey(idOrKey);
+            entityId = projection.getId();
```

Comment:
this resolving entityId by key it's copy/paste, pls extract method to avoid copy paste, and use this method in other places too, pls, very pls =))

Reply:
fixed